### ISSUE TITLE
[wg/aria] Accessible Rich Internet Applications Working Group Recharter

### ISSUE THREAD

**Issue by daniel-montalvo** [w3c/strategy#466]
New charter proposal, reviewers please take note.

# Charter Review

[Accessible Rich Internet Applications:](https://w3c.github.io/charter-drafts/2024/aria-charter.html)

[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts/2024/aria-charter.html)

If applicable:
[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2022%2F02%2Faria-charter&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2024%2Faria-charter.html)

[chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=wg/aria)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* New
 - [ ] New WG 
 - [ ] New IG
 - If this is a new WG or IG charter request, link to [Advance Notice](), and any issue discussion: 

* Existing 
 - [x] Existing WG recharter 
 - [ ] Existing IG recharter
 - If this is a charter extension or revision, any issue discussion:

Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/w3c/strategy/projects/2).**

Communities suggested for outreach: 

Known or potential areas of concern: 

Where would charter proponents like to see issues raised? (this strategy funnel issue)

Anything else we should think about as we review? 

Note: proposed chairs should be copied  @spectranaut  @jnurthen on this issue.

**Comment by svgeesus** [comment]
Any reason not to request horizontal review of the proposed charter?

**Comment by plehegar** [comment]
SVG-AAM has to be explicitly a deliverable of the WG. At the moment, ARIA is just meant to collaborate with the SVG WG, not to deliver it. Shouldn't we list at the same level as the HTML AAM, and add that it is a joint deliverable?

cc @iadawn

**Comment by daniel-montalvo** [comment]
@plehegar 

SVG-AAM is now explicitly added as a deliverable.
https://w3c.github.io/charter-drafts/2024/aria-charter.html

**Comment by simoneonofri** [comment]
Hi @daniel-montalvo

I read the Security Considerations of the various deliverables. I noticed that it often says:

> This specification introduces no new security considerations.

Could you explain the reasoning behind these statements? Not because I think it is wrong but because I would like to understand. 

Thank you in advance :)

Simone

**Comment by daniel-montalvo** [comment]
Hi @simoneonofri 

There was a discussion about Privacy and Seccurity issues that the ARIA spec could introduced back in 2020. For background, see
[Detectability of assistive technologies](https://github.com/w3c/aria/issues/1371)

The group reached an agreement that this was more a privacy issue, and that it could happen in ARIA just as it could happen with other technologies.

As a conclusion, the following text was added in the [privacy section of the ARIA spec](https://www.w3.org/TR/wai-aria-1.3/#privacy-considerations).

<blockquote>
<p>In accordance with <a href="https://www.w3.org/TR/design-principles/#do-not-expose-use-of-assistive-tech">Web Platform Design Principles</a>, this specification provides no programmatic interface to determine if information is being used by Assistive Technologies. However, this specification does allow an author to present different information to users of Assistive Technologies from the information available to users who do not use Assistive Technologies. This is possible using many features of the ARIA specification, just as this is possible using many other parts of the web technology stack. This content disparity could be abused to perform <a href="https://www.w3.org/TR/fingerprinting-guidance/#active-0">active fingerprinting</a> of users of Assistive Technologies.</p>
</blockquote>

Because changes in the ARIA specs only add, qualify, or deprecate the use of roles, atrributes, and properties, the group believes that these changes introduce "no new security considerations".

Is there anything else you are missing for the security sections?

**Comment by simoneonofri** [comment]
Hi @daniel-montalvo,


Thank you very much for the clarification and context.

Two potential reflections in the area of security:
 - If the attributes/element can be used to trigger events on JavaScript (e.g., as the `onlick` attribute can be) then it might make sense to specify that there is an increased attack surface for e.g. Cross Site Scripting.
 - Another issue - but it's a generalized problem still found even on HTML5 elements/attributes in 2024 - is that any ant-XSS blocklist filters (and blocklist in fact are not a very good security practice) do not also include WAI-ARIA attributes/elements that can be used to escape.

Thank you,

Simone

**Comment by ruoxiran** [comment]
no comment or request from APA.

**Comment by daniel-montalvo** [comment]
Hi.

Thanks for sharing your thoughts.

> * If the attributes/element can be used to trigger events on JavaScript (e.g., as the `onlick` attribute can be) then it might make sense to specify that there is an increased attack surface for e.g. Cross Site Scripting.

None of these attributes are used to trigger JavaScript events.

> * Another issue - but it's a generalized problem still found even on HTML5 elements/attributes in 2024 - is that any ant-XSS blocklist filters (and blocklist in fact are not a very good security practice) do not also include WAI-ARIA attributes/elements that can be used to escape.

That sounds like something that will affect more specs than just ARIA. I am open to following along through your advice on how to write security considerations and update ARIA ones later on based on this (and potentially other) thoughts.

**Comment by himorin** [comment]
no comment or request from i18n

**Comment by plehegar** [comment]
PING is fine

**Comment by daniel-montalvo** [comment]
Closing. Review is complete. TAG does not intend to review this.

**Comment by himorin** [comment]
reopening this, this is not just for horizontal review, but through whole chartering period.

**Comment by plehegar** [comment]
I submitted a pull request. See https://github.com/w3c/charter-drafts/pull/589 .

In addition:
- In external organization: "IMS Global Learning Consortium" does no longer exist.
- Charter history hasn't been updated to reflect the draft charter

**Comment by plehegar** [comment]
Note that, in my pull request, I removed the Web Platform WG from the coordination section since that W3C does no longer exist. For HTML, thew coordination with the WHATG is already listed.

**Comment by svgeesus** [comment]
Thank you @plehegar for spotting these issues, which I came here to report, and also for a PR to fix them.

In general this rechartering has not been done well. Instead of starting from the current charter template and adding up-to-date info, it seems to be a light warming-over of an existing charter. This produces a lot of extra work for horizontal reviewers, for TiLT, and (if everything is not fixed) for the AC as well.

**Comment by siusin** [comment]
See https://github.com/w3c/charter-drafts/pull/597. 

Wonder if the ARIA WG would consider adding some specific description of their work about the [HTML Accessibility API Mappings](https://www.w3.org/TR/html-aam-1.0/) spec and the [ARIA in HTML](https://www.w3.org/TR/html-aria/) spec in the Scope section.

**Comment by daniel-montalvo** [comment]
Thanks @svgeesus 

>     * In external organization: "IMS Global Learning Consortium" does no longer exist.

w3c/charter-drafts#599 fixes this

>     * Charter history hasn't been updated to reflect the draft charter

Do we have  guidance as to how we should be indicating this?
Should this be in the last row of the Charter History table? Somewhere else?

**Comment by plehegar** [comment]
> > * Charter history hasn't been updated to reflect the draft charter
> > ```
> 
> Do we have guidance as to how we should be indicating this? Should this be in the last row of the Charter History table? Somewhere else?

Yes please. Add row in the table to note the substantial changes.

**Comment by svgeesus** [comment]
1. Under [Success Criteria](https://w3c.github.io/charter-drafts/2024/aria-charter.html#success-criteria), while I do appreciate copying some text from [CSSWG charter: Success Criteria](https://www.w3.org/2023/01/css-2023.html#success-criteria), you should change "module" to "Specification" because CSS WG has one deliverable with many modules while ARIA does not.

2. The charter template has

> Consider adding this clause if the Group does not intend to move to REC: All new features should have expressions of interest from at least two potential implementors before being incorporated in the specification.

This has been removed, and yet nine deliverables are marked **(living standard)**. The sucess criteria uses only the wording about moving to Proposed Rec, which only applies to two deliverables.

3. The template has

> Consider adopting a healthy testing policy, such as: To promote interoperability, all changes made to specifications in Candidate Recommendation or to features that have deployed implementations should have [tests](https://www.w3.org/2019/02/testing-policy.html). Testing efforts should be conducted via the [Web Platform Tests](https://github.com/web-platform-tests/wpt) project.

That text has all been removed. Do ARIA specifications not need tests? Is WPT not suitable for ARIA tests?

4. The template has

> This (Working|Interest) Group expects to follow the TAG [Web Platform Design Principles](https://www.w3.org/TR/design-principles/).

which has been removed. Does ARIA not intend to follow those principles? Or do they not apply?

5. In [Communication](https://w3c.github.io/charter-drafts/2024/aria-charter.html#communication) the link to [Accessible Rich Internet Applications Working Group home page.](https://www.w3.org/WAI/about/groups/ariawg/) goes to https://www.w3.org/WAI/about/groups/ariawg/ but should instead go to https://www.w3.org/groups/wg/aria

By the way, I notice the change from "Each specification should contain sections detailing all known security and privacy implications" to "Each specification should contain **separate** sections detailing all known security and privacy implications". Nice addition, which I have [also made to the template](https://github.com/w3c/charter-drafts/commit/9e7b0d3d0bc775fb7119fe01f6617ca37b951f78)

**Comment by daniel-montalvo** [comment]
> > > * Charter history hasn't been updated to reflect the draft charter
> > 
> > 
> > Do we have guidance as to how we should be indicating this? Should this be in the last row of the Charter History table? Somewhere else?
> 
> Yes please. Add row in the table to note the substantial changes.

Added table row in [11.1](https://w3c.github.io/charter-drafts/2024/aria-charter.html#history) and detailed changelog in [11.2](https://w3c.github.io/charter-drafts/2024/aria-charter.html#changelog)

**Comment by svgeesus** [comment]
What does

> PDF Accessibility API Mappings 1.0 (living standard)
    TBD

mean? I know that means To Be Determined, but shouldn't it say "No Draft"?

In general the charter is looking much better now.

**Comment by plehegar** [comment]
one minor nit: the change log section is meant to contain the changes **after** the charter is approved (there is an [HTML comment in the charter template](https://github.com/w3c/charter-drafts/blob/gh-pages/charter-template.html#L513) explaining this). The section could be left as-is for the AC Review but will need to be emptied once the change is approved, to avoid future confusion.

**Comment by daniel-montalvo** [comment]
@svgeesus 

w3c/charter-drafts#605 fixes this.

**Comment by svgeesus** [comment]
> [w3c/charter-drafts#605](https://github.com/w3c/charter-drafts/pull/605) fixes this.

That gets rid of the TBD. But it should also say:

> Draft state: No draft

**Comment by daniel-montalvo** [comment]
@svgeesus 
Added "no draft" and milestone at https://github.com/w3c/charter-drafts/pull/606

**Comment by daniel-montalvo** [comment]
Charter available in W3C space at
https://www.w3.org/2025/01/aria-charter.html

**Comment by npdoty** [comment]
Since the privacy review was completed, we've chartered a Privacy Working Group taking over the work from PING (Privacy Interest Group). If editorial changes are still accepted, we could make that update. But no objection or concern -- I'm confident no one would be long confused by the older link.

**Comment by daniel-montalvo** [comment]
Thanks @npdoty 

#622 should fix this.

@plehegar I think this passes for minor changes under [Managing changes to charters after review](https://www.w3.org/Guide/process/charter.html#manage-changes) and we could include it.

However, if this is going to trigger another Call for Review, I would prefer not to. include #622 Otherwise I'd be happy to. I assume at some point https://www.w3.org/groups/ig/privacy/ will clearly indicate that work is now happening at https://www.w3.org/groups/wg/privacy/

**Comment by daniel-montalvo** [comment]
See Charter questionnaire results at https://www.w3.org/2002/09/wbs/33280/aria-2025-charter-review/results

**Comment by plehegar** [comment]
> Thanks @npdoty
> 
> #622 should fix this.
> 
> @plehegar I think this passes for minor changes under [Managing changes to charters after review](https://www.w3.org/Guide/process/charter.html#manage-changes) and we could include it.

agreed.

**Comment by svgeesus** [comment]
> . I assume at some point https://www.w3.org/groups/ig/privacy/ will clearly indicate that work is now happening at https://www.w3.org/groups/wg/privacy/

Good point, and I agree that it should.

**Comment by plehegar** [comment]
Group as [announced](https://lists.w3.org/Archives/Public/public-new-work/2024Dec/0001.html).
